### PR TITLE
App: in PropertyPythonObject::fromString only assign a dict to __dict…

### PR DESCRIPTION
**Pull request by wwmayer**
…__ attribute

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [ ]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9012">pipeline status</a> for feature branch [PR_9012](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9012). Pipeline [816307798](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/816307798) was triggered at [cf6a37e](https://github.com/FreeCAD/FreeCAD/pull/9012/commits/cf6a37e306818dcb9f2504bdfe625213af78881c). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/App/PropertyPythonObject.cpp [hunk 1]
```diff
@@ -136,7 +136,9 @@ void PropertyPythonObject::fromString(const std::string& repr)
             state.apply(args);
         }
         else if (this->object.hasAttr("__dict__")) {
-            this->object.setAttr("__dict__", res);
+            if (res.isDict()) {
+                this->object.setAttr("__dict__", res);
+            }
```
Comment (chennes) [inline src/App/PropertyPythonObject.cpp:141]: No need for an `else` clause here to handle the error condition?

Comment (wwmayer) [reply src/App/PropertyPythonObject.cpp:141]: Changed it to `!res.isNone()`

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9012">pipeline status</a> for feature branch [PR_9012](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9012). Pipeline [816985617](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/816985617) was triggered at [3477290](https://github.com/FreeCAD/FreeCAD/pull/9012/commits/347729090a511946a0af1d08b4bf9080c4775440). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/App/PropertyPythonObject.cpp (hunk 1 above)
Comment (chennes) [reply src/App/PropertyPythonObject.cpp:141]: Thanks for the explanation, makes sense to me.

**Comment by wwmayer** [comment]
Can we merge this or are there any concerns?

**Pull request merged**